sql/thriftserver/pom.xml

```diff
@@ -0,0 +1,153 @@
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.spark</groupId>
+        <artifactId>spark-parent_2.12</artifactId>
+        <version>3.0.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>spark-thriftserver_2.12</artifactId>
+    <packaging>jar</packaging>
+    <name>Spark Project Thrift Server</name>
+    <url>http://spark.apache.org/</url>
+    <properties>
+        <sbt.project.name>thriftserver</sbt.project.name>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-hive_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-hive_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-cli</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-jdbc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-beeline</artifactId>
+        </dependency>
```

Comment:
The hive- dependencies are only for beeline now, and for testing with Hive JDBC client?

Reply:
> The hive- dependencies are only for beeline now, and for testing with Hive JDBC client?

Current beeline still need these dependencies. In first version we can't do all the thing, so copy this from `sql/hive-thriftserver` . First step is to replace old one.

Reply:
Maybe we could consider separating the server and the client in separate modules? A `sql/jdbc` module would have `hive-beeline` and `hive-jdbc` and `hive-service` and be used to provide `bin/beeline`, but it will only be a testing dependency of thriftserver, to provide the client for testing - that would further separate the server code from Hive. Could then the existing sql/hive-thriftserver also be depending on it?
@wangyum @gatorsmile what do you think? (definitely in a separate PR from this one)

Reply:
+1